client/my-sites/checkout/composite-checkout/components/wp-checkout.tsx

```diff
@@ -297,6 +299,10 @@ export default function WPCheckout( {
 
 	const isDIFMInCart = hasDIFMProduct( responseCart );
 
+	if ( hasPremiumPlan( responseCart ) ) {
+		loadExperimentAssignment( 'calypso_plans_postpurchasepage_newdesign_202301_v1' );
```

Comment:
I don't know enough to be sure; is it safe to call `loadExperimentAssignment()` on every render?

Reply:
Good point, @sirbrillig. I've investigated a bit and it looks ok. As it's mentioned also in the [package readme](https://github.com/Automattic/wp-calypso/tree/trunk/packages/explat-client#usage):
> Call as many times and as much as you like, we manage the state and requests.

So even if the preloading function was designed to be used outside of the React application (signup controller was the first usage), it should be safe to call it on multiple renders.

But in a React Component any side effect should be explicitly defined, together with its dependencies. So I've wrapped the call in a `useEffect` and placed it before any of the component returns. There shouldn't be any noticeable change in behavior. The function will still get called multiple times when a new `responseCart` object is passed.